auth/auth.go

```diff
@@ -34,6 +33,8 @@ type Auth interface {
 	Options() Options
 	// Generate a new account
 	Generate(id string, opts ...GenerateOption) (*Account, error)
+	// Login to an existing account
+	Login(id string, opts ...LoginOption) (*Account, error)
```

Comment:
I think if we're going to have a Login method it should explicitly ask for id, secret as args rather than as an option. Otherwise seeing this I'm just seeing a mirror of the Token method in which case:

`Token(id string, opts ...TokenOption)` where Secret or RefreshToken are passed in as options and we operate based on whats provided. Feels cleaner? Then `micro login` is only calling the Token method and passing in secret or token as required.

Reply:
The reason I done it like this is so that oauth providers can login an account without a password.

Reply:
After thinking about this, I agree. I'll merge those two methods.

Reply:
I'm just feeling like the methods are redundant in the sense that Login/Token are basically a mirror of each other. Whether you need to login or exchange a token its basically one and the same with different params. So being able to specific Secret or Token as an option to the Token method would allow it in one place.

The other thing I'm thinking:

```
type Token struct {
    // Refresh is a function which renews the token
    Refresh func() (*Token, error)
}
```

Rather than specifying the refresh token maybe we specify a refresh function instead which abstracts it away. So when we call auth.Token the auth provider sets up the Refresh() function to handle the details.

Reply:
Actually, for oauth logic we wouldn't have either a refresh token (since they're logging in), or a secret. I'll have to add an edgecase to allow unauthenticated token generation if the account has a provider, such as I did with Login.

Reply:
In this case Refresh function would be nil. Go does similar things in certain structs where they just do a nil check on a field and if its not present they ignore it, otherwise its executed. I think it cleans up the logic a bit.

Reply:
I'm not sure about this since it breaks down for non go usage, e.g. when I pass my token to the frontend or CLI I can't later call Refresh.

Reply:
When you call the Auth.Token rpc method you'll get back a refresh token. It's left to the frontend to handle its own logic around refreshing (likely by calling the token method again with the refresh token). From solely a go-micro standpoint if you're talking just about interfaces and usage in code its just cleaner but I can share a refactor later.